src/streamlink/plugins/cmmedia.py

```diff
@@ -0,0 +1,133 @@
+            validate.any(None, validate.all(
+                validate.transform(self.iframe_url_re.search),
+                validate.any(None, validate.all(validate.get(1), validate.url())),
+            )),
+        ))
+
+        if not iframe_url:
+            return
+
+        m = self.partner_ids_re.search(iframe_url)
+        if not m:
+            log.error("Failed to find partner IDs in IFRAME URL")
+
+        p = m.group(1)
+        sp = m.group(2)
+
+        json = self.session.http.get(iframe_url, schema=validate.Schema(
+            validate.transform(self.json_re.search),
+            validate.any(None, validate.all(
+                validate.get(1),
+                validate.transform(self.unescape_quotes),
+                validate.parse_json(),
+                validate.any({"error": str}, {
+                    "entryResult": {
+                        "contextData": {
+                            "isSiteRestricted": bool,
+                            "isCountryRestricted": bool,
+                            "isSessionRestricted": bool,
+                            "isIpAddressRestricted": bool,
+                            "isUserAgentRestricted": bool,
+                            "flavorAssets": [{
+                                "id": str,
+                            }],
+                        },
+                        "meta": {
+                            "id": str,
+                            "name": str,
+                            "categories": validate.any(None, str),
+                        },
+                    },
+                }),
+            )),
+        ))
+
+        if not json:
+            return
+
+        if "error" in json:
+            log.error(f"API error: {json['error']}")
+            return
+
+        json = json.get("entryResult")
+
+        if self.is_restricted(json["contextData"]):
+            return
+
+        self.id = json["meta"]["id"]
+        self.title = json["meta"]["name"]
+        author = self.session.http.get(self.url, schema=validate.Schema(
+            validate.parse_html(),
+            validate.xml_xpath_string(".//h1[@class='btn-title']/text()"),
+        ))
+        if author:
+            self.author = author
+        if json["meta"]["categories"]:
+            self.category = json["meta"]["categories"]
+
+        base_url = f"https://cdnapisec.kaltura.com/p/{p}/sp/{sp}/playManifest/entryId/{json['meta']['id']}/flavorIds/"
+        trailing_path = "/protocol/https/a.m3u8"
+
+        for asset in json["contextData"]["flavorAssets"]:
+            yield from HLSStream.parse_variant_playlist(
+                self.session,
+                f"{base_url}{asset['id']}/format/applehttp{trailing_path}",
+                name_fmt="{pixels}_{bitrate}",
+            ).items()
+
+            yield from DASHStream.parse_manifest(
+                self.session,
+                f"{base_url}{asset['id']}/format/mpegdash{trailing_path}",
+            ).items()
```

Comment:
I did wonder about this as well.  I have not seen any streams where HLS is unavailable, so using HLS and dropping DASH would make most sense, as it can be displayed and ordered by bit rate and pixels.